src/replication.c

```diff
@@ -2698,6 +2698,7 @@ static int dualChannelReplHandleEndOffsetResponse(connection *conn, sds *err) {
     /* Initiate repl_provisional_primary to act as this replica temp primary until RDB is loaded */
     server.repl_provisional_primary.conn = server.repl_transfer_s;
     memcpy(server.repl_provisional_primary.replid, primary_replid, CONFIG_RUN_ID_SIZE);
+    server.repl_provisional_primary.replid[CONFIG_RUN_ID_SIZE] = '\0';
```

Comment:
```suggestion
    memcpy(server.repl_provisional_primary.replid, primary_replid, sizeof(server.repl_provisional_primary.replid));
```

I'm partial to this, which seems to be the convention used elsewhere, to make sure we're copying the entire buffer. Then it's less likely for these two lines to get split.

This one is static storage class, so it'll always be '\0' so this one should be fine.

Reply:
I think the codebase is a bit split, e.g. see https://github.com/valkey-io/valkey/blob/5ea9b52e7395db48aa0bae9410b3e2363964bee2/src/replication.c#L3134

But I agree with your logic here with using sizeof.

And you're right, this one wouldn't be a bug so long as it zeroed out on initialization